app/workers/moderator/unpublish_all_articles_worker.rb

```diff
@@ -11,6 +11,9 @@ def perform(user_id)
       user.articles.published.find_each do |article|
         if article.has_frontmatter?
           article.body_markdown.sub!(/^published:\s*true\s*$/, "published: false")
+        else
+          # nullify published_at when unpublishing for the articles published in a rich markdown editor
+          article.published_at = nil
```

Comment:
Until I saw this, I did not realize that we supported setting `published` in frontmatter! Should our unpublish strategy also rewrite this frontmatter as well? (If so, then that's possibly an additional reason we might look at encapsulating (un)publishing in a setter on the model.)

Reply:
@jaw6 I mentioned this concern in the pr description (sections "Concerns"). I think that ideally we should change frontmatter on unpublishing articles from editor v1 as well.
As for the updating `published_at` logic, I mentioned my concerns and plans for the refactoring in the [inital pr](https://github.com/forem/forem/pull/17939/files#r901483566). However, I wouldn't add this logic to the model, because it's already overloaded with responsibilities, but rather manage separately.

Reply:
:+1: That all makes sense, I should have seen that earlier!